dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/domain/Time.kt

```diff
@@ -6,7 +6,19 @@
 
 package com.datadog.android.core.internal.domain
 
+import java.util.concurrent.TimeUnit
+
 internal data class Time(
     val timestamp: Long = System.currentTimeMillis(),
     val nanoTime: Long = System.nanoTime()
 )
+
+internal fun Long.asTime(): Time {
+    // Because nanoTime only measures the nanoseconds since the beginning
+    // of the current JVM lifetime, we need to approximate the nanotime we want.
+    // We simply convert the delay between the desired and real timestamp and
+    // apply it to the measured nanotime
+    val now = Time()
+    val offset = this - now.timestamp
+    return Time(this, TimeUnit.MILLISECONDS.toNanos(offset) + now.nanoTime)
```

Comment:
Hmm it is a bit the other way around. I guess in order to make it more clear you could compute the offset as : 
`now.timestamp-this` and the substract it.

Reply:
It all boils down to the same result, and I find both as clear (or unclear) as one another